src/addBusinessDays/index.ts

```diff
@@ -54,25 +70,69 @@ export default function addBusinessDays(
 
   // Get remaining days not part of a full week
   let restDays = Math.abs(amount % businessDays.length)
-
   // Loops over remaining days
   while (restDays > 0) {
     date.setDate(date.getDate() + sign)
-    if (!isHoliday(date)) restDays -= 1
+    if (!isNonWorkingDay(date)) restDays -= 1
+  }
+
+  const filterExceptions = (exceptionString: string) => {
+    const exceptionDate = new Date(exceptionString)
+    const [earlierDate, laterDate] =
+      sign === 1 ? [dirtyDate, date] : [date, dirtyDate]
+    if (
+      (isBefore(exceptionDate, laterDate) &&
+        isAfter(exceptionDate, earlierDate)) ||
+      isSameDay(exceptionDate, laterDate) ||
+      isSameDay(exceptionDate, earlierDate)
+    ) {
+      if (
+        exceptions[exceptionString] === true &&
+        !businessDays.includes(exceptionDate.getDay())
+      ) {
+        return true
+      } else if (
+        exceptions[exceptionString] === false &&
+        businessDays.includes(exceptionDate.getDay())
+      ) {
+        return true
+      }
```

Comment:
thought (non-blocking): Looks like this logic can be simplified?
```suggestion
      if (
        exceptions[exceptionString] === !businessDays.includes(exceptionDate.getDay())
      ) {
        return true
      } 
```

Also generally, it feels like a few strategically placed comments explaining how this changes here work could be helpful.

Reply:
Yeah that's a lot more clean, I'll update!